internal/tools/cloudsqlpg/cloudsqlpgprecheckmajorversionupgrade/cloudsqlpgprecheckmajorversionupgrade.go

```diff
@@ -127,35 +142,36 @@ func (t Tool) Invoke(ctx context.Context, params tools.ParamValues, accessToken
 	if !ok || project == "" {
 		return nil, fmt.Errorf("missing or empty 'project' parameter")
 	}
-	instanceName, ok := paramsMap["name"].(string)
+	instanceName, ok := paramsMap["instance"].(string)
 	if !ok || instanceName == "" {
-		return nil, fmt.Errorf("missing or empty 'name' parameter")
+		return nil, fmt.Errorf("missing or empty 'instance' parameter")
 	}
 	targetVersion, ok := paramsMap["targetDatabaseVersion"].(string)
 	if !ok || targetVersion == "" {
+		// This should not happen due to the default value
 		return nil, fmt.Errorf("missing or empty 'targetDatabaseVersion' parameter")
 	}
 
 	service, err := t.Source.GetService(ctx, string(accessToken))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get HTTP client from source: %w", err)
 	}
 
-	// Prepare the request body
 	reqBody := &sqladmin.InstancesPreCheckMajorVersionUpgradeRequest{
 		PreCheckMajorVersionUpgradeContext: &sqladmin.PreCheckMajorVersionUpgradeContext{
 			TargetDatabaseVersion: targetVersion,
 		},
 	}
 
-	// Call the PreCheckMajorVersionUpgrade API
 	call := service.Instances.PreCheckMajorVersionUpgrade(project, instanceName, reqBody).Context(ctx)
 	op, err := call.Do()
 	if err != nil {
 		return nil, fmt.Errorf("failed to start pre-check operation: %w", err)
 	}
 
-	// Poll the Long-Running Operation
+	const pollTimeout = 5 * time.Minute
```

Comment:
5 minutes is a long time to poll without any response to the user. Is this the experience we want? We do have a get_operations tool that works with the create instances and other tools that just return an operation id.